prow/cmd/webhook-server/helpers.go

```diff
@@ -246,37 +254,183 @@ func createValidatingWebhookConfig(ctx context.Context, caPem string, client ctr
+		return fmt.Errorf("failed to get validating webhook config: %w", err)
 	}
 	oldValidatingWebhook := validatingWebhookConfig.DeepCopy()
 	validatingWebhookConfig.Webhooks[0].ClientConfig.CABundle = []byte(caPem)
 	if err := client.Patch(ctx, &validatingWebhookConfig, ctrlruntimeclient.MergeFrom(oldValidatingWebhook), patchOptions); err != nil {
-		return fmt.Errorf("failed to patch validation webhook config: %w", err)
+		return fmt.Errorf("failed to patch validating webhook config: %w", err)
+	}
+	return nil
+}
+
+func createMutatingWebhookConfig(ctx context.Context, caPem string, client ctrlruntimeclient.Client) error {
+	operations := []admregistration.OperationType{"CREATE"}
+	scope := admregistration.ScopeType("*")
+	path := "/mutate"
+	sideEffects := admregistration.SideEffectClass("None")
+
+	mutatingWebhookConfig := &admregistration.MutatingWebhookConfiguration{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "MutatingWebhookConfiguration",
+			APIVersion: "admissionregistration.k8s.io/v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: prowJobMutatingWebhookName,
+		},
+		Webhooks: []admregistration.MutatingWebhook{
+			{
+				Name: prowJobMutatingWebhookName,
+				ObjectSelector: &v1.LabelSelector{
+					MatchLabels: map[string]string{
+						"admission-webhook": "enabled",
+						"default-me":        "enabled",
+					},
+				},
+				Rules: []admregistration.RuleWithOperations{
+					{
+						Operations: operations,
+						Rule: admregistration.Rule{
+							APIGroups:   []string{"prow.k8s.io"},
+							APIVersions: []string{"v1"},
+							Resources:   []string{"prowjobs"},
+							Scope:       &scope,
+						},
+					},
+				},
+				ClientConfig: admregistration.WebhookClientConfig{
+					Service: &admregistration.ServiceReference{
+						Namespace: defaultNamespace,
+						Name:      prowjobAdmissionServiceName,
+						Path:      &path,
+					},
+					CABundle: []byte(caPem),
+				},
+				SideEffects:             &sideEffects,
+				AdmissionReviewVersions: []string{"v1"},
+			},
+		},
+	}
+
+	createOptions := &ctrlruntimeclient.CreateOptions{
+		FieldManager: "webhook-server",
+	}
+
+	err := client.Create(ctx, mutatingWebhookConfig, createOptions)
+	if err != nil && strings.Contains(err.Error(), configAlreadyExistsError) {
+		logrus.Info("MutatingWebhookConfiguration already exists, proceeding to patch")
+		if err := patchMutatingWebhookConfig(ctx, caPem, client); err != nil {
+			return fmt.Errorf("failed to patch mutating webhook config: %w", err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("failed to create mutating webhook config: %w", err)
+	}
+
+	return nil
+}
+
+func patchMutatingWebhookConfig(ctx context.Context, caPem string, client ctrlruntimeclient.Client) error {
+	key := types.NamespacedName{
+		Namespace: defaultNamespace,
+		Name:      prowJobMutatingWebhookName,
+	}
+
+	patchOptions := &ctrlruntimeclient.PatchOptions{
+		FieldManager: "webhook-server",
```

Comment:
this is already set in `create` function, no need to do it again I guess?

Reply:
patching is a different operation which requires its own set of optional options

Reply:
not entirely sure but sounds like this is something has to be set, and this option needs to be the same as the one used when create, if that's the case then this needs to be a const